js_modules/dagit/packages/ui/src/components/MiddleTruncate.tsx

```diff
@@ -1,22 +1,46 @@
 import useResizeObserver from '@react-hook/resize-observer';
-import debounce from 'lodash/debounce';
 import * as React from 'react';
 import styled from 'styled-components/macro';
 
 import {calculateMiddleTruncation} from './calculateMiddleTruncation';
 
-interface Props {
-  text: string;
-}
-
-const getFont = (element: HTMLDivElement) => getComputedStyle(element).font;
+/**
+ * A component that performs middle truncation on a given string, based on the evaluated width
+ * of a container div.
+ *
+ * The component will render the provided string -- with no height/visibility -- into our target
+ * container to determine the maximum available width. This width value and the computed font
+ * style are then used to determine the longest middle-truncated string that can fit within the width.
+ *
+ * When the DOM element resizes, the measurement and calculation steps will occur again.
+ */
+export const MiddleTruncate: React.FC<{text: string}> = React.memo(({text}) => {
+  // Track the font style and target maximum width. `null` means no measurement has
+  // taken place yet.
+  const [truncatedText, setTruncatedText] = React.useState<string | null>(null);
```

Comment:
Now only tracking truncated text in state, not the style/width values -- these are now only tracked within the calculation function, which is the only place we need them.